### PR TITLE
Add new variable for the spark-env.sh template

### PR DESCRIPTION
**Pull request by danilo404**
Add new variable to allow the possibility to set up a different path for the hdfs native libraries path.

### CODE REVIEW

**Comment by hectcastro** [comment]
Hey @d-riecken, I just wanted to let you know that we've allocated some time to review your PR this week or next (so that you didn't feel like we were leaving you hanging).

Also, thanks for taking the time to submit this PR back upstream.

File: defaults/main.yml [hunk 1]
```diff
@@ -7,10 +7,13 @@ spark_src_dir: "/usr/local/src"
 spark_conf_dir: "/etc/spark"
 spark_usr_parent_dir: "/usr/lib"  #this is the folder where the spark archive will be extracted
 spark_usr_dir: "/usr/lib/spark"   #this is the symlink to the extracted/installed spark
+spark_usr_dir_orig: "{{spark_usr_parent_dir}}/spark-{{spark_version}}"
```
Comment (tnation14) [inline defaults/main.yml:10]: This directory is referenced throughout the role; for consistency's sake, could you update other references to use `spark_usr_dir_orig`?

File: defaults/main.yml [hunk 2]
```diff
@@ -24,6 +27,21 @@ spark_shims_enabled: yes
 spark_hive_metastore_db_installed: false
 spark_hive_site_properties: {}
 
+spark_extra_libs_fileset: []     # Optional list of files to deploy to the spark lib folder
+# Note: before v2. the dir is 'lib' instead of 'jars'
+spark_extra_libs_dir: "{{spark_usr_dir_orig +'/'+ ('jars' if is_spark_min_v2 else 'lib') }}"
+spark_extra_libs_dir: "{{ spark_usr_dir_orig }}/jars"
```
Comment (tnation14) [inline defaults/main.yml:33]: This is a duplicate of the variable above.

File: tasks/spark-services.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+- set_fact:
```
Comment (tnation14) [inline tasks/spark-services.yml:1]: While the functionality here seems useful to your specific use of Spark, I'd like to limit `azavea.spark` to installing/configuring a base Spark distribution. If we start installing additional Spark services through this role, I'm concerned that we'll incur a higher level of complexity/maintenance burden in 2 areas:

- Adding Upstart & SysVInit startup scripts means we'll have to be on the lookout for any changes to the service managers _and_ changes to the commands to start up the Spark services.

- Installing services through this role means that we are also responsible for configuring each service. That can mean more templated config files (which are subject to change), setting environment variables, etc.

I think this functionality is better suited for a `spark-services` wrapper role that depends on `azavea.spark`.

**Review by tnation14 — COMMENTED** [review]
Hi @d-riecken, thanks for your contribution! Due to some of the concerns listed below, I can't merge all of the commits in this PR, but if you could open a smaller PR containing 2b20b288e1fa62d19da8d1b7e8bf189f9d620b50 I'd be happy to merge that. Thanks!

File: tasks/spark-services.yml (hunk 3 above)
Comment (danilo404) [reply tasks/spark-services.yml:1]: You are right, thanks!

File: defaults/main.yml (hunk 1 above)
Comment (danilo404) [reply defaults/main.yml:10]: Hi tnation14, sorry for the late reply, I'm organising our repo and will close this PR and come back with a more consistent PR later with all the fixes you proposed. Thanks for the review!

File: defaults/main.yml (hunk 2 above)
Comment (danilo404) [reply defaults/main.yml:33]: Fixed, thanks! See my first comment.

File: defaults/main.yml (hunk 1 above)
Comment (tnation14) [reply defaults/main.yml:10]: No problem, I look forward to the next PR. Thanks again for the contribution 👌🏾

**Pull request closed without merging**